x/dex/keeper/pair.go

```diff
@@ -30,13 +30,21 @@ func (k Keeper) GetPairCount(ctx sdk.Context, contractAddr string) uint64 {
 	return binary.BigEndian.Uint64(cnt)
 }
 
-func (k Keeper) AddRegisteredPair(ctx sdk.Context, contractAddr string, pair types.Pair) {
+func (k Keeper) AddRegisteredPair(ctx sdk.Context, contractAddr string, pair types.Pair) bool {
+	// Only add pairs that have not been added before
+	for _, prevPair := range k.GetAllRegisteredPairs(ctx, contractAddr) {
```

Comment:
instead of iterating here, I think we should make the `AddRegisteredPair` idempotent since if we are trying to add multiple pairs, itll be super poor performance since we'd iterate over all registered pairs  for each one.

Reply:
Discussed offline, we don't need to address this at this moment, we can optimize this when we move to a string representation of denom. along with that, we'd want to do a store migration for the registered pair store and have the registered pair key be build by string literal prefix + contract address + pair literal. This we we can check for existence without iteration.
Here is the location of the key we would need to modify: https://github.com/sei-protocol/sei-chain/blob/a15402115078923698cafb36295def41761d8754/x/dex/types/keys.go#L73

When we do the modification, we would need to do a store migration to ensure the old data format converted.

CC @Kbhat1 @codchen

Reply:
Yup, will do this when doing the larger PR to update the price type from enum to string (for arbitrary token denoms)